### PR TITLE
docs: refactor sidebar menu

### PR DESCRIPTION
**Pull request by thepassle**
Bunch of changes to the docs. This would focus a bit more on the separation between developing an app, or developing a web component. Made this PR so its easier to click through and discuss, and see if its a good idea or not.

Currently, a lot of the information is the same (linting/testing for example) for apps and web components. Maybe this is confusing for users? It does however give a bit more freedom for the future, (for example: we can have different publishing recommendations for `web-component` and `app`) but the question is if thats worth it.

Also moved some of the tooling under a `tooling` section. I need to still check if I moved all the tooling, and if I moved the right tooling.

I also updated to the latest version of vuepress

### CODE REVIEW

File: docs/apps/testing/testing-karma-bs.md [hunk 1]
```diff
@@ -0,0 +1 @@
+/Users/Passle/open-wc/packages/testing-karma-bs/README.md
```
Comment (thepassle) [inline docs/apps/testing/testing-karma-bs.md]:
```suggestion
../../../packages/testing-karma-bs/README.md
```

File: docs/apps/testing/testing-karma.md [hunk 2]
```diff
@@ -0,0 +1 @@
+/Users/Passle/open-wc/packages/testing-karma/README.md
```
Comment (thepassle) [inline docs/apps/testing/testing-karma.md]:
```suggestion
../../../packages/testing-karma/README.md
```

File: docs/apps/testing/testing-wallaby.md [hunk 3]
```diff
@@ -0,0 +1 @@
+/Users/Passle/open-wc/packages/testing-wallaby/README.md
```
Comment (thepassle) [inline docs/apps/testing/testing-wallaby.md]:
```suggestion
../../../packages/testing-wallaby/README.md
```

File: docs/apps/testing/testing.md [hunk 4]
```diff
@@ -0,0 +1 @@
+/Users/Passle/open-wc/packages/testing/README.md
```
Comment (thepassle) [inline docs/apps/testing/testing.md]:
```suggestion
../../../packages/testing/README.md
```

File: docs/web-component/linting/README.md [hunk 5]
```diff
@@ -0,0 +1 @@
+/Users/Passle/open-wc/docs/linting/README.md
```
Comment (thepassle) [inline docs/web-component/linting/README.md]:
```suggestion
../../docs/linting/README.md
```

File: docs/web-component/linting/linting-eslint.md [hunk 6]
```diff
@@ -0,0 +1 @@
+/Users/Passle/open-wc/packages/eslint-config/README.md
```
Comment (thepassle) [inline docs/web-component/linting/linting-eslint.md]:
```suggestion
../../../packages/eslint-config/README.md
```

File: docs/web-component/linting/linting-prettier.md [hunk 7]
```diff
@@ -0,0 +1 @@
+/Users/Passle/open-wc/packages/prettier-config/README.md
```
Comment (thepassle) [inline docs/web-component/linting/linting-prettier.md]:
```suggestion
../../../packages/prettier-config/README.md
```

File: docs/web-component/linting/linting-types.md [hunk 8]
```diff
@@ -0,0 +1 @@
+/Users/Passle/open-wc/docs/linting/linting-types.md
```
Comment (thepassle) [inline docs/web-component/linting/linting-types.md]:
```suggestion
../../docs/linting/linting-types.md
```

File: docs/web-component/testing/testing-karma-bs.md [hunk 9]
```diff
@@ -0,0 +1 @@
+/Users/Passle/open-wc/packages/testing-karma-bs/README.md
```
Comment (thepassle) [inline docs/web-component/testing/testing-karma-bs.md]:
```suggestion
../../../packages/testing-karma-bs/README.md
```

File: docs/web-component/testing/testing-karma.md [hunk 10]
```diff
@@ -0,0 +1 @@
+/Users/Passle/open-wc/packages/testing-karma/README.md
```
Comment (thepassle) [inline docs/web-component/testing/testing-karma.md]:
```suggestion
../../../packages/testing-karma/README.md
```

File: docs/web-component/testing/testing-wallaby.md [hunk 11]
```diff
@@ -0,0 +1 @@
+/Users/Passle/open-wc/packages/testing-wallaby/README.md
```
Comment (thepassle) [inline docs/web-component/testing/testing-wallaby.md]:
```suggestion
../../../packages/testing-wallaby/README.md
```

File: docs/web-component/testing/testing.md [hunk 12]
```diff
@@ -0,0 +1 @@
+/Users/Passle/open-wc/packages/testing/README.md
```
Comment (thepassle) [inline docs/web-component/testing/testing.md]:
```suggestion
../../../packages/testing/README.md
```

**Comment by bennypowers** [comment]
Do we have a PR preview?

**Comment by thepassle** [comment]
Yes here: https://deploy-preview-947--open-wc-org.netlify.com/

Still WIP though (I forgot to move `automating`). And as I mentioned above, currently theres a _lot_ of duplicate information (testing/linting most notably)

**Comment by daKmoR** [comment]
hmm the submenus (General, Application, ...) do not open for me 🙈

**Comment by thepassle** [comment]
Hmmm weird, they do open for me:

Screenshot 2019-11-04 at 20 39 59

**Comment by thepassle** [comment]
@daKmoR made a new deploy: https://5dc07f6add2ebaebc64cf855--open-wc-org.netlify.com/guide/

File: docs/.vuepress/config.js [hunk 13]
```diff
@@ -1,12 +1,134 @@
...
+        collapsable: true,
+        children: [['/web-component/building/', 'Getting started']],
+      },
+      {
+        title: 'Testing',
+        collapsable: true,
+        children: [
+          ['/web-component/testing/', 'Getting started'],
+          '/web-component/testing/testing',
+          '/web-component/testing/testing-karma',
```
Comment (thepassle) [inline docs/.vuepress/config.js]: Make these:

`['/web-component/testing/testing-karma', 'Testing Karma']`

File: docs/.vuepress/config.js [hunk 14]
```diff
@@ -1,12 +1,134 @@
...
+      {
+        title: 'Testing',
+        collapsable: true,
+        children: [
+          ['/apps/testing/', 'Getting started'],
+          '/apps/testing/testing',
+          '/apps/testing/testing-karma',
+          '/apps/testing/testing-sinon',
+          '/apps/testing/testing-karma-bs',
+          '/apps/testing/testing-wallaby',
```
Comment (thepassle) [inline docs/.vuepress/config.js]: Make these:

`['/web-component/testing/testing-karma', 'Testing Karma']`

File: docs/.vuepress/config.js [hunk 15]
```diff
@@ -1,12 +1,134 @@
...
+      {
+        title: 'Testing',
+        collapsable: true,
+        children: [
+          ['/web-component/testing/', 'Getting started'],
+          '/web-component/testing/testing',
+          '/web-component/testing/testing-karma',
+          '/web-component/testing/testing-sinon',
+          '/web-component/testing/testing-karma-bs',
+          '/web-component/testing/testing-wallaby',
```
Comment (thepassle) [inline docs/.vuepress/config.js]: Make these:

`['/web-component/testing/testing-karma', 'Testing Karma']`

**Comment by LarsDenBakker** [comment]
I originally proposed the refactor of the menus. The primary reason for this, is that it's hard to distinguish between content/guides for apps and content/guides for reusable component projects. 

Also, some of the menus like testing are all very diffrerent and I think it's unclear for a new user to know what's what.

What if we split up the website into three different sections. Probably not as double nested menus, but as top level navigation. Similar to how webpack does multiple sections for guides, api, plugins, concepts etc. https://webpack.js.org/guides/

**Guides & docs**
- Guides
  - Codelabs
  - Code examples
- How-to & FAQ
  - How to test events
- Architecture & libraries
  - Routing
  - State management 

**Configs & recommendations**
- Linting
  - Getting started
  - ESLint
  - Prettier
  - Types
- Developing
  - Getting started
- Testing
  - Getting started
  - Karma
  - Testing libs (just the basics to get started, deep dive in tools pages)
  - Browserstack
- Demoing
  - Getting started
- Building apps for production
  - Getting started
  - Rollup
  - Webpack
- Publishing reusable components
  - Getting started

**Tools**
- Developing
  - es-dev-server
- Testing
  - karma-esm
  - chai-a11y-axe
  - semantic dom diff
  - testing-helpers
- Building
  - rollup-plugin-index-html
  - webpack-index-html-plugin

This is just a rough draft, specifics can be worked out.

**Comment by LarsDenBakker** [comment]
I originally proposed the refactor of the menus. The primary reason for this, is that it's hard to distinguish between content/guides for apps and content/guides for reusable component projects. 

Also, some of the menus like testing are all very diffrerent and I think it's unclear for a new user to know what's what.

What if we split up the website into three different sections. Probably not as double nested menus, but as top level navigation. Similar to how webpack does multiple sections for guides, api, plugins, concepts etc. https://webpack.js.org/guides/

**Guides & docs**
- Guides
  - Codelabs
  - Code examples
- How-to & FAQ
  - How to test events
- Architecture & libraries
  - Routing
  - State management 

**Configs & recommendations**
- Linting
  - Getting started
  - ESLint
  - Prettier
  - Types
- Developing
  - Getting started
- Testing
  - Getting started
  - Karma
  - Testing libs (just the basics to get started, deep dive in tools pages)
  - Browserstack
- Demoing
  - Getting started
- Building apps for production
  - Getting started
  - Rollup
  - Webpack
- Publishing reusable components
  - Getting started

**Tools**
- Developing
  - es-dev-server
- Testing
  - karma-esm
  - chai-a11y-axe
  - semantic dom diff
  - testing-helpers
- Building
  - rollup-plugin-index-html
  - webpack-index-html-plugin

This is just a rough draft, specifics can be worked out.

**Comment by thepassle** [comment]
Yes I like that. I think the idea of having separate sections for app/wc is really nice, but right now its not really working. Maybe it will at some point in the future

**Comment by LarsDenBakker** [comment]
Well after thinking about it, we really only need the split for building. I did add that split in there now.

**Pull request closed without merging**